department-of-veterans-affairs/next-build
# initial refactor of media image handling + tests

## Description
Closes #[15485](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15485).

## Testing done
`yarn dev` and noticed the greatly reduced amount of noise from failed requests

`yarn test` and noticed that all tests are passing with the simplified formatter for media

## Screenshots
Notice the only failed request comes from the feature_toggles work loaded in from vets-websites assets. Much easier to ignore. 😁 
```
- wait compiling...
- event compiled client and server successfully in 192 ms (547 modules)
Failed request (Attempt 1 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
Failed request (Attempt 2 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
Failed request (Attempt 3 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
Failed request (Attempt 4 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
Failed request (Attempt 5 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
Failed request (Attempt 6 of 6): https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/router/translate-path?path=%2Fv0%2Ffeature_toggles
- error src/lib/drupal/drupalClient.ts (48:12) @ wrappedCrossFetch
- error Error: Failed request
    at wrappedCrossFetch (webpack-internal:///./src/lib/drupal/drupalClient.ts:45:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async RetryOperation._fn (file:///Users/heffner/dev/next-build/node_modules/p-retry/index.js:55:13) {
  digest: undefined
}
  46 |         }`
  47 |       )
> 48 |       throw new Error('Failed request')
     |            ^
  49 |     }
  50 |     return response
  51 |   }
- wait compiling...
- event compiled successfully in 79 ms (143 modules)
- wait compiling...
- event compiled successfully in 95 ms (143 modules)
- wait compiling...
- event compiled client and server successfully in 249 ms (547 modules)
```

## QA steps
Run yarn dev locally
visit any story listing path and notice images rendering without spamming your terminal.

```[tasklist]
- [ ] Images correctly render on story listing and individual news story pages
```

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:
